addons/website_sale/models/sale_order.py

```diff
@@ -50,15 +50,29 @@ def _compute_abandoned_cart(self):
             order.is_abandoned_cart = bool(domain)
 
     def _search_abandoned_cart(self, operator, value):
-        abandoned_delay = self.website_id and self.website_id.cart_abandoned_delay or 1.0
-        abandoned_datetime = fields.Datetime.to_string(datetime.utcnow() - relativedelta(hours=abandoned_delay))
-        abandoned_domain = expression.normalize_domain([
-            ('date_order', '<=', abandoned_datetime),
-            ('team_id.team_type', '=', 'website'),
-            ('state', '=', 'draft'),
-            ('partner_id', '!=', self.env.ref('base.public_partner').id),
-            ('order_line', '!=', False)
-        ])
+        query = '''
+        SELECT main.id
+        FROM(
+            SELECT 
+                so.id,
+                so.date_order + COALESCE(web.cart_abandoned_delay, 1) * INTERVAL '1 hour' AS date_abandoned
+            FROM
+                sale_order so
+                LEFT JOIN website web ON (web.id = so.website_id)
+                LEFT JOIN res_users public_user ON (public_user.id = web.user_id)
+                LEFT JOIN crm_team team ON (team.id = so.team_id)
+            WHERE
+                team.team_type = 'website'
+                AND so.state = 'draft'
+                AND so.date_order IS NOT NULL
+                AND so.website_id IS NOT NULL
+                AND so.partner_id != public_user.partner_id
+            ) AS main
+        WHERE
+            main.date_abandoned < Now()'''
```

Comment:
Maybe we could avoid nesting, it is already readable enough.
Although a comment on why sql is needed might be useful
```sql
            SELECT
                so.id
            FROM
                sale_order so
                LEFT JOIN website web ON (web.id = so.website_id)
                LEFT JOIN res_users public_user ON (public_user.id = web.user_id)
                LEFT JOIN crm_team team ON (team.id = so.team_id)
            WHERE
                team.team_type = 'website'
                AND so.state = 'draft'
                AND so.date_order IS NOT NULL
                AND so.website_id IS NOT NULL
                AND so.partner_id != public_user.partner_id
                AND so.date_order + COALESCE(web.cart_abandoned_delay, 1) * INTERVAL '1 hour' < Now()
 ```